CRM/Contact/BAO/ContactType.php

```diff
@@ -871,14 +869,17 @@ protected static function getAllContactTypes() {
     $cacheKey = 'all_' . $GLOBALS['tsLocale'];
     $contactTypes = $cache->get($cacheKey);
     if ($contactTypes === NULL) {
-      $contactTypes = (array) ContactType::get(FALSE)
-        ->setSelect(['id', 'name', 'label', 'description', 'is_active', 'is_reserved', 'image_URL', 'parent_id', 'parent_id:name', 'parent_id:label'])
-        ->execute()->indexBy('name');
-
+      $query = CRM_Utils_SQL_Select::from('civicrm_contact_type')
+        ->select(['id', 'name', 'label', 'description', 'is_active', 'is_reserved', 'image_URL', 'parent_id']);
+      $dao = CRM_Core_DAO::executeQuery($query->toSQL());
+      $contactTypes = array_column($dao->fetchAll(), NULL, 'name');
+      $name_options = self::buildOptions('parent_id', 'validate');
+      $label_options = self::buildOptions('parent_id', 'get');
       foreach ($contactTypes as $id => $contactType) {
-        $contactTypes[$id]['parent'] = $contactType['parent_id:name'];
-        $contactTypes[$id]['parent_label'] = $contactType['parent_id:label'];
-        unset($contactTypes[$id]['parent_id:name'], $contactTypes[$id]['parent_id:label']);
+        if ($contactType['parent_id']) {
+          $contactTypes[$id]['parent'] = $name_options[$contactType['parent_id']];
+          $contactTypes[$id]['parent_label'] = $label_options[$contactType['parent_id']];
```

Comment:
To fix the failing test, add something to the loop to cast some values to the correct type (int or bool)